doc/adr/0015-current-API-authentication-strategy.md

```diff
@@ -0,0 +1,116 @@
+
+Configuration is primarily via the CLI but the OIDC method can also read configuration from a secret (by default `oidc-auth`) which over-rides CLI configuration. The `OIDC.TokenDuration` attribute is used to set all cookie expiry times (not just those related to OIDC).
+
+It is important to note that the authentication method is also 'used' by the `/oauth2/userinfo` endpoint. This endpoint searches cookies and headers to find principal information and returns it in an JSON object. This does not use the `PrincipalGetter` interface and discovers principal data independently.
+
+### Authentication from the server to Kubernetes
+
+The Gitops server itself only needs a very limited set of permissions (`impersonate`, `get,list` on the secrets used to configure authentication and `get,list` on namespaces). All interaction with the Kubernetes API should use the `impersonate` method to request data using the user's principal rather than Gitops'.
+
+The principal returned by `WithAPIAuthentication` is added to the request context to authenticate with the Kubernetes API. For OIDC and user-account methods the principal is extracted from the header or cookie token which is expected to have the following attributes:
+
+```javascript
+{
+  "email": "someone@example.invalid",  // required
+  "groups": ["team-a", "team-b"],      // optional
+}
+```
+
+The `email` and `groups` terms are then used to configure [impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation) to the Kubernetes API which allows the gitops' service account to act on the user's behalf.
+
+When token-passthrough is used the user-supplied token is treated as a bearer token and no-impersonation is carried out. It is assumed that the Kubernetes API will know how to extract any relevant Principal information from it directly.
+
+Communication to the Kubernetes API should be done via the client returned by `GetImpersonatedClient` which will configure impersonation. There are a couple of places in `pkg/server/auth/` and in `cmd/gitops-server/cmd/cmd.go` where a raw (i.e. non-impersonated) client are used, this is done to read secrets during initialisation and to call the `TokenReview` APIs that authenticate with pass-through tokens.
+
+## Consequences
+
+This is not an exhaustive list of consequences of the above but aims to capture important points.
```

Comment:
this section to me kind of points out that we need 
- to ensure that us developing need to have this into consideration ... so comes to my mind on how to materialize this expectation ?
- to ensure that our users have enough guidance around the topic to configure (if required) the product appropriately? do you manage to know where in the user documentation this is gathered?

Reply:
>     * to ensure that us developing need to have this into consideration ... so comes to my mind on how to materialize this expectation ?

In terms of developing I think some work can be done to improve the architecture to reduce the risks listed here (e.g. #2582). I'd also hope that this ADR and similar work can be circulated/brought to developer's attention if they need to work in these areas so they're better informed.

>     * to ensure that our users have enough guidance around the topic to configure (if required) the product appropriately? do you manage to know where in the user documentation this is gathered?

There's a fair bit of this [documented](https://docs.gitops.weave.works/docs/configuration/securing-access-to-the-dashboard/) for the end users already. I don't know if you think there's something obvious missing there?

Reply:
>In terms of developing I think some work can be done to improve the architecture to reduce the risks listed here (e.g. https://github.com/weaveworks/weave-gitops/issues/2582).

I definitely see what you mean and makes sense as you did to raise tickets for enhancements. 

What is the priority you think the story has in terms of the risk for the products?